server/application_api: TestStatusAPICombinedStatementsWithFullScans failed

server/application_api.TestStatusAPICombinedStatementsWithFullScans [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21323329?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21323329?buildTab=artifacts#/) on master @ [064776dd40c32b9f38f52f5728fa5941ada5561d](https://github.com/cockroachdb/cockroach/commits/064776dd40c32b9f38f52f5728fa5941ada5561d):


```
=== RUN   TestStatusAPICombinedStatementsWithFullScans
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/bdcf2ab810bc38279e9dce2da8e9dabd/logTestStatusAPICombinedStatementsWithFullScans4161333895
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    sql_stats_test.go:811: 
        	Error Trace:	pkg/server/application_api/sql_stats_test.go:811
        	            				pkg/server/application_api/sql_stats_test.go:852
        	Error:      	Should be true
        	Test:       	TestStatusAPICombinedStatementsWithFullScans
        	Messages:   	Expected statement 'SELECT name FROM players WHERE age < _' not found in response: {"statements":[{"key":{"key_data":{"query":"CREATE TABLE players (id INT8 PRIMARY KEY, name STRING, \"position\" STRING, age INT8, goals INT8)","app":"TestCombinedStatementsWithFullScans","distSQL":false,"implicit_txn":false,"vec":false,"full_scan":false,"database":"football","plan_hash":0,"query_summary":"CREATE TABLE players (id INT8 PRIMARY KEY, name STRING, \"position\" STRING, age INT8, goals INT8)","transaction_fingerprint_id":0},"aggregated_ts":"2026-04-22T19:00:00Z","aggregation_interval":0},"id":1651796105171899064,"stats":{"count":1,"first_attempt_count":1,"max_retries":0,"legacy_last_err":"","legacy_last_err_redacted":"","num_rows":{"mean":0,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"parse_lat":{"mean":0.000049169,"squared_diffs":0},"plan_lat":{"mean":0.170737408,"squared_diffs":0},"run_lat":{"mean":0.168739641,"squared_diffs":0},"service_lat":{"mean":0.33953008,"squared_diffs":0},"overhead_lat":{"mean":0.000003862,"squared_diffs":0},"sensitive_info":{"last_err":"","most_recent_plan_description":{"name":""},"most_recent_plan_timestamp":"0001-01-01T00:00:00Z"},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"rows_written":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":1719706,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"sql_type":"TypeDDL","last_exec_timestamp":"2026-04-22T19:59:45.465945671Z","nodes":[3],"regions":["test"],"plan_gists":["AiXSAQ=="],"latency_info":{"min":0.33953008,"max":0.33953008,"p50":0,"p90":0,"p99":0},"last_error_code":"","failure_count":0,"used_follower_read":false,"generic_count":0,"stmt_hints_count":0,"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0},"canary_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}},"stable_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}}},"txn_fingerprint_ids":[0]},{"key":{"key_data":{"query":"DROP INDEX IF EXISTS idx_age","app":"TestCombinedStatementsWithFullScans","distSQL":false,"implicit_txn":false,"vec":false,"full_scan":false,"database":"football","plan_hash":0,"query_summary":"DROP INDEX IF EXISTS idx_age","transaction_fingerprint_id":0},"aggregated_ts":"2026-04-22T19:00:00Z","aggregation_interval":0},"id":3779827305370537204,"stats":{"count":1,"first_attempt_count":1,"max_retries":0,"legacy_last_err":"","legacy_last_err_redacted":"","num_rows":{"mean":0,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"parse_lat":{"mean":0.000025531,"squared_diffs":0},"plan_lat":{"mean":0.292033183,"squared_diffs":0},"run_lat":{"mean":0.000161572,"squared_diffs":0},"service_lat":{"mean":0.292223637,"squared_diffs":0},"overhead_lat":{"mean":0.000003351,"squared_diffs":0},"sensitive_info":{"last_err":"","most_recent_plan_description":{"name":""},"most_recent_plan_timestamp":"0001-01-01T00:00:00Z"},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"rows_written":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":22440,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"sql_type":"TypeDDL","last_exec_timestamp":"2026-04-22T19:59:47.010882558Z","nodes":[3],"regions":["test"],"plan_gists":["Ais="],"latency_info":{"min":0.292223637,"max":0.292223637,"p50":0,"p90":0,"p99":0},"last_error_code":"","failure_count":0,"used_follower_read":false,"generic_count":0,"stmt_hints_count":0,"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0},"canary_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}},"stable_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}}},"txn_fingerprint_ids":[0]},{"key":{"key_data":{"query":"GRANT  SYSTEM VIEWACTIVITY TO authentic_user_noadmin","app":"","distSQL":false,"implicit_txn":false,"vec":false,"full_scan":false,"database":"defaultdb","plan_hash":0,"query_summary":"GRANT  SYSTEM VIEWACTIVITY TO authentic_user_noadmin","transaction_fingerprint_id":0},"aggregated_ts":"2026-04-22T19:00:00Z","aggregation_interval":0},"id":10411270073732320383,"stats":{"count":1,"first_attempt_count":1,"max_retries":0,"legacy_last_err":"","legacy_last_err_redacted":"","num_rows":{"mean":0,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"parse_lat":{"mean":0.000021146,"squared_diffs":0},"plan_lat":{"mean":0.000283719,"squared_diffs":0},"run_lat":{"mean":0.600113095,"squared_diffs":0},"service_lat":{"mean":0.600422802,"squared_diffs":0},"overhead_lat":{"mean":0.000004842,"squared_diffs":0},"sensitive_info":{"last_err":"","most_recent_plan_description":{"name":""},"most_recent_plan_timestamp":"0001-01-01T00:00:00Z"},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"rows_written":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":2248784,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":33007,"squared_diffs":0}},"sql_type":"TypeDCL","last_exec_timestamp":"2026-04-22T19:59:44.080599315Z","nodes":[3],"regions":["test"],"plan_gists":["Ais="],"latency_info":{"min":0.600422802,"max":0.600422802,"p50":0,"p90":0,"p99":0},"last_error_code":"","failure_count":0,"used_follower_read":false,"generic_count":0,"stmt_hints_count":0,"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0},"canary_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}},"stable_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}}},"txn_fingerprint_ids":[0]},{"key":{"key_data":{"query":"SELECT avg(goals) FROM players","app":"TestCombinedStatementsWithFullScans","distSQL":true,"implicit_txn":false,"vec":false,"full_scan":true,"database":"football","plan_hash":0,"query_summary":"SELECT avg(goals) FROM players","transaction_fingerprint_id":0},"aggregated_ts":"2026-04-22T19:00:00Z","aggregation_interval":0},"id":11648331846954514632,"stats":{"count":1,"first_attempt_count":1,"max_retries":0,"legacy_last_err":"","legacy_last_err_redacted":"","num_rows":{"mean":1,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"parse_lat":{"mean":0.00005845,"squared_diffs":0},"plan_lat":{"mean":0.000316688,"squared_diffs":0},"run_lat":{"mean":0.001867632,"squared_diffs":0},"service_lat":{"mean":0.002251986,"squared_diffs":0},"overhead_lat":{"mean":0.000009216,"squared_diffs":0},"sensitive_info":{"last_err":"","most_recent_plan_description":{"name":""},"most_recent_plan_timestamp":"0001-01-01T00:00:00Z"},"bytes_read":{"mean":115,"squared_diffs":0},"rows_read":{"mean":2,"squared_diffs":0},"rows_written":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":192,"squared_diffs":0},"max_mem_usage":{"mean":51200,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":3,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":110737,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":2,"squared_diffs":0},"step_count_internal":{"mean":2,"squared_diffs":0},"seek_count":{"mean":1,"squared_diffs":0},"seek_count_internal":{"mean":1,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":85,"squared_diffs":0},"point_count":{"mean":2,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"sql_type":"TypeDML","last_exec_timestamp":"2026-04-22T19:59:45.94041473Z","nodes":[1,3],"kv_node_ids":[1],"regions":["test"],"plan_gists":["AgHUAQIAEAAAAAwGAg=="],"indexes":["106@1"],"latency_info":{"min":0.002251986,"max":0.002251986,"p50":0,"p90":0,"p99":0},"last_error_code":"","failure_count":0,"used_follower_read":false,"generic_count":0,"stmt_hints_count":0,"kv_cpu_time_nanos":{"mean":211954,"squared_diffs":0},"canary_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}},"stable_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}}},"txn_fingerprint_ids":[0]},{"key":{"key_data":{"query":"SET database = football","app":"TestCombinedStatementsWithFullScans","distSQL":false,"implicit_txn":false,"vec":false,"full_scan":false,"database":"football","plan_hash":0,"query_summary":"SET database = football","transaction_fingerprint_id":0},"aggregated_ts":"2026-04-22T19:00:00Z","aggregation_interval":0},"id":12425643783624206004,"stats":{"count":1,"first_attempt_count":1,"max_retries":0,"legacy_last_err":"","legacy_last_err_redacted":"","num_rows":{"mean":0,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"parse_lat":{"mean":0.000018896,"squared_diffs":0},"plan_lat":{"mean":0.00009235,"squared_diffs":0},"run_lat":{"mean":0.260495418,"squared_diffs":0},"service_lat":{"mean":0.26060983,"squared_diffs":0},"overhead_lat":{"mean":0.000003166,"squared_diffs":0},"sensitive_info":{"last_err":"","most_recent_plan_description":{"name":""},"most_recent_plan_timestamp":"0001-01-01T00:00:00Z"},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"rows_written":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":986364,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"sql_type":"TypeDML","last_exec_timestamp":"2026-04-22T19:59:44.986109338Z","nodes":[3],"regions":["test"],"plan_gists":["Ais="],"latency_info":{"min":0.26060983,"max":0.26060983,"p50":0,"p90":0,"p99":0},"last_error_code":"","failure_count":0,"used_follower_read":false,"generic_count":0,"stmt_hints_count":0,"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0},"canary_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}},"stable_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}}},"txn_fingerprint_ids":[0]},{"key":{"key_data":{"query":"CREATE DATABASE football","app":"TestCombinedStatementsWithFullScans","distSQL":false,"implicit_txn":false,"vec":false,"full_scan":false,"database":"defaultdb","plan_hash":0,"query_summary":"CREATE DATABASE football","transaction_fingerprint_id":0},"aggregated_ts":"2026-04-22T19:00:00Z","aggregation_interval":0},"id":14235364449065395993,"stats":{"count":1,"first_attempt_count":1,"max_retries":0,"legacy_last_err":"","legacy_last_err_redacted":"","num_rows":{"mean":0,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"parse_lat":{"mean":0.000020757,"squared_diffs":0},"plan_lat":{"mean":0.508018036,"squared_diffs":0},"run_lat":{"mean":0.002323763,"squared_diffs":0},"service_lat":{"mean":0.510366536,"squared_diffs":0},"overhead_lat":{"mean":0.00000398,"squared_diffs":0},"sensitive_info":{"last_err":"","most_recent_plan_description":{"name":""},"most_recent_plan_timestamp":"0001-01-01T00:00:00Z"},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"rows_written":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":2138608,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"sql_type":"TypeDDL","last_exec_timestamp":"2026-04-22T19:59:44.725137527Z","nodes":[3],"regions":["test"],"plan_gists":["Ais="],"latency_info":{"min":0.510366536,"max":0.510366536,"p50":0,"p90":0,"p99":0},"last_error_code":"","failure_count":0,"used_follower_read":false,"generic_count":0,"stmt_hints_count":0,"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0},"canary_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}},"stable_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}}},"txn_fingerprint_ids":[0]},{"key":{"key_data":{"query":"INSERT INTO players(id, name, \"position\", age, goals) VALUES (_, __more__), (__more__)","app":"TestCombinedStatementsWithFullScans","distSQL":false,"implicit_txn":false,"vec":false,"full_scan":false,"database":"football","plan_hash":0,"query_summary":"INSERT INTO players(id, name, \"posi...)","transaction_fingerprint_id":0},"aggregated_ts":"2026-04-22T19:00:00Z","aggregation_interval":0},"id":15994326564351379676,"stats":{"count":1,"first_attempt_count":1,"max_retries":0,"legacy_last_err":"","legacy_last_err_redacted":"","num_rows":{"mean":2,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"parse_lat":{"mean":0.000070309,"squared_diffs":0},"plan_lat":{"mean":0.411658622,"squared_diffs":0},"run_lat":{"mean":0.054968556,"squared_diffs":0},"service_lat":{"mean":0.466701945,"squared_diffs":0},"overhead_lat":{"mean":0.000004458,"squared_diffs":0},"sensitive_info":{"last_err":"","most_recent_plan_description":{"name":""},"most_recent_plan_timestamp":"0001-01-01T00:00:00Z"},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"rows_written":{"mean":2,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":493262,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"sql_type":"TypeDML","last_exec_timestamp":"2026-04-22T19:59:45.933189334Z","nodes":[3],"regions":["test"],"plan_gists":["AiAE1AEB"],"latency_info":{"min":0.466701945,"max":0.466701945,"p50":0,"p90":0,"p99":0},"last_error_code":"","failure_count":0,"used_follower_read":false,"generic_count":0,"stmt_hints_count":0,"kv_cpu_time_nanos":{"mean":281266,"squared_diffs":0},"canary_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}},"stable_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}}},"txn_fingerprint_ids":[0]},{"key":{"key_data":{"query":"SET application_name = _","app":"","distSQL":false,"implicit_txn":false,"vec":false,"full_scan":false,"database":"defaultdb","plan_hash":0,"query_summary":"SET application_name = _","transaction_fingerprint_id":0},"aggregated_ts":"2026-04-22T19:00:00Z","aggregation_interval":0},"id":16494915433690409681,"stats":{"count":1,"first_attempt_count":1,"max_retries":0,"legacy_last_err":"","legacy_last_err_redacted":"","num_rows":{"mean":0,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"parse_lat":{"mean":0.000021695,"squared_diffs":0},"plan_lat":{"mean":0.000124188,"squared_diffs":0},"run_lat":{"mean":0.000193126,"squared_diffs":0},"service_lat":{"mean":0.000342043,"squared_diffs":0},"overhead_lat":{"mean":0.000003034,"squared_diffs":0},"sensitive_info":{"last_err":"","most_recent_plan_description":{"name":""},"most_recent_plan_timestamp":"0001-01-01T00:00:00Z"},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"rows_written":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":23962,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"sql_type":"TypeDML","last_exec_timestamp":"2026-04-22T19:59:44.086035827Z","nodes":[3],"regions":["test"],"plan_gists":["Ais="],"latency_info":{"min":0.000342043,"max":0.000342043,"p50":0,"p90":0,"p99":0},"last_error_code":"","failure_count":0,"used_follower_read":false,"generic_count":0,"stmt_hints_count":0,"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0},"canary_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}},"stable_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}}},"txn_fingerprint_ids":[0]},{"key":{"key_data":{"query":"SELECT name FROM players WHERE age \u003e= _","app":"TestCombinedStatementsWithFullScans","distSQL":true,"implicit_txn":false,"vec":false,"full_scan":true,"database":"football","plan_hash":0,"query_summary":"SELECT name FROM players","transaction_fingerprint_id":0},"aggregated_ts":"2026-04-22T19:00:00Z","aggregation_interval":0},"id":17078876116494672480,"stats":{"count":1,"first_attempt_count":1,"max_retries":0,"legacy_last_err":"","legacy_last_err_redacted":"","num_rows":{"mean":2,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"parse_lat":{"mean":0.000050516,"squared_diffs":0},"plan_lat":{"mean":0.000372064,"squared_diffs":0},"run_lat":{"mean":0.003414506,"squared_diffs":0},"service_lat":{"mean":0.003846718,"squared_diffs":0},"overhead_lat":{"mean":0.000009632,"squared_diffs":0},"sensitive_info":{"last_err":"","most_recent_plan_description":{"name":""},"most_recent_plan_timestamp":"0001-01-01T00:00:00Z"},"bytes_read":{"mean":115,"squared_diffs":0},"rows_read":{"mean":2,"squared_diffs":0},"rows_written":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":376,"squared_diffs":0},"max_mem_usage":{"mean":40960,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":4,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":197910,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":2,"squared_diffs":0},"step_count_internal":{"mean":2,"squared_diffs":0},"seek_count":{"mean":1,"squared_diffs":0},"seek_count_internal":{"mean":1,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":85,"squared_diffs":0},"point_count":{"mean":2,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"sql_type":"TypeDML","last_exec_timestamp":"2026-04-22T19:59:45.94481627Z","nodes":[1,3],"kv_node_ids":[1],"regions":["test"],"plan_gists":["AgHUAQIACgAAAAMFAgYC"],"indexes":["106@1"],"latency_info":{"min":0.003846718,"max":0.003846718,"p50":0,"p90":0,"p99":0},"last_error_code":"","failure_count":0,"used_follower_read":false,"generic_count":0,"stmt_hints_count":0,"kv_cpu_time_nanos":{"mean":234243,"squared_diffs":0},"canary_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}},"stable_stats":{"count":0,"run_lat":{"mean":0,"squared_diffs":0},"plan_lat":{"mean":0,"squared_diffs":0}}},"txn_fingerprint_ids":[0]}],"last_reset":"2026-04-22T19:59:34.464299564Z","internal_app_name_prefix":"$ internal","transactions":[{"stats_data":{"statement_fingerprint_ids":[12425643783624206004],"app":"TestCombinedStatementsWithFullScans","stats":{"count":1,"max_retries":0,"num_rows":{"mean":0,"squared_diffs":0},"service_lat":{"mean":0.260758879,"squared_diffs":0},"retry_lat":{"mean":0,"squared_diffs":0},"commit_lat":{"mean":0.000015001,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":986364,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"rows_written":{"mean":0,"squared_diffs":0},"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0}},"aggregated_ts":"2026-04-22T19:00:00Z","transaction_fingerprint_id":221657573235331435,"aggregation_interval":0}},{"stats_data":{"statement_fingerprint_ids":[11648331846954514632],"app":"TestCombinedStatementsWithFullScans","stats":{"count":1,"max_retries":0,"num_rows":{"mean":1,"squared_diffs":0},"service_lat":{"mean":0.002406384,"squared_diffs":0},"retry_lat":{"mean":0,"squared_diffs":0},"commit_lat":{"mean":0.000013693,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"bytes_read":{"mean":115,"squared_diffs":0},"rows_read":{"mean":2,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":192,"squared_diffs":0},"max_mem_usage":{"mean":51200,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":3,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":110737,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":2,"squared_diffs":0},"step_count_internal":{"mean":2,"squared_diffs":0},"seek_count":{"mean":1,"squared_diffs":0},"seek_count_internal":{"mean":1,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":85,"squared_diffs":0},"point_count":{"mean":2,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"rows_written":{"mean":0,"squared_diffs":0},"kv_cpu_time_nanos":{"mean":211954,"squared_diffs":0}},"aggregated_ts":"2026-04-22T19:00:00Z","transaction_fingerprint_id":1064130812164067095,"aggregation_interval":0}},{"stats_data":{"statement_fingerprint_ids":[10411270073732320383],"app":"","stats":{"count":1,"max_retries":0,"num_rows":{"mean":0,"squared_diffs":0},"service_lat":{"mean":1.195206884,"squared_diffs":0},"retry_lat":{"mean":0,"squared_diffs":0},"commit_lat":{"mean":0.068380769,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":2248784,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":33007,"squared_diffs":0}},"rows_written":{"mean":0,"squared_diffs":0},"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0}},"aggregated_ts":"2026-04-22T19:00:00Z","transaction_fingerprint_id":4548635297285012384,"aggregation_interval":0}},{"stats_data":{"statement_fingerprint_ids":[17078876116494672480],"app":"TestCombinedStatementsWithFullScans","stats":{"count":1,"max_retries":0,"num_rows":{"mean":2,"squared_diffs":0},"service_lat":{"mean":0.003986871,"squared_diffs":0},"retry_lat":{"mean":0,"squared_diffs":0},"commit_lat":{"mean":0.000012227,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"bytes_read":{"mean":115,"squared_diffs":0},"rows_read":{"mean":2,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":376,"squared_diffs":0},"max_mem_usage":{"mean":40960,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":4,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":197910,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":2,"squared_diffs":0},"step_count_internal":{"mean":2,"squared_diffs":0},"seek_count":{"mean":1,"squared_diffs":0},"seek_count_internal":{"mean":1,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":85,"squared_diffs":0},"point_count":{"mean":2,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"rows_written":{"mean":0,"squared_diffs":0},"kv_cpu_time_nanos":{"mean":234243,"squared_diffs":0}},"aggregated_ts":"2026-04-22T19:00:00Z","transaction_fingerprint_id":4785046135168357823,"aggregation_interval":0}},{"stats_data":{"statement_fingerprint_ids":[16494915433690409681],"app":"","stats":{"count":1,"max_retries":0,"num_rows":{"mean":0,"squared_diffs":0},"service_lat":{"mean":0.002306594,"squared_diffs":0},"retry_lat":{"mean":0,"squared_diffs":0},"commit_lat":{"mean":0.000024426,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":23962,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"rows_written":{"mean":0,"squared_diffs":0},"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0}},"aggregated_ts":"2026-04-22T19:00:00Z","transaction_fingerprint_id":5443175735277133070,"aggregation_interval":0}},{"stats_data":{"statement_fingerprint_ids":[14235364449065395993],"app":"TestCombinedStatementsWithFullScans","stats":{"count":1,"max_retries":0,"num_rows":{"mean":0,"squared_diffs":0},"service_lat":{"mean":0.638659184,"squared_diffs":0},"retry_lat":{"mean":0,"squared_diffs":0},"commit_lat":{"mean":0.12784374,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":2138608,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"rows_written":{"mean":0,"squared_diffs":0},"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0}},"aggregated_ts":"2026-04-22T19:00:00Z","transaction_fingerprint_id":7704974824955039942,"aggregation_interval":0}},{"stats_data":{"statement_fingerprint_ids":[15994326564351379676],"app":"TestCombinedStatementsWithFullScans","stats":{"count":1,"max_retries":0,"num_rows":{"mean":2,"squared_diffs":0},"service_lat":{"mean":0.466826133,"squared_diffs":0},"retry_lat":{"mean":0,"squared_diffs":0},"commit_lat":{"mean":0.061673463,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":493262,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"rows_written":{"mean":2,"squared_diffs":0},"kv_cpu_time_nanos":{"mean":281266,"squared_diffs":0}},"aggregated_ts":"2026-04-22T19:00:00Z","transaction_fingerprint_id":8256503328528383747,"aggregation_interval":0}},{"stats_data":{"statement_fingerprint_ids":[3779827305370537204],"app":"TestCombinedStatementsWithFullScans","stats":{"count":1,"max_retries":0,"num_rows":{"mean":0,"squared_diffs":0},"service_lat":{"mean":0.292351244,"squared_diffs":0},"retry_lat":{"mean":0,"squared_diffs":0},"commit_lat":{"mean":0.00001191,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":22440,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"rows_written":{"mean":0,"squared_diffs":0},"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0}},"aggregated_ts":"2026-04-22T19:00:00Z","transaction_fingerprint_id":11175429196517178155,"aggregation_interval":0}},{"stats_data":{"statement_fingerprint_ids":[1651796105171899064],"app":"TestCombinedStatementsWithFullScans","stats":{"count":1,"max_retries":0,"num_rows":{"mean":0,"squared_diffs":0},"service_lat":{"mean":0.479558761,"squared_diffs":0},"retry_lat":{"mean":0,"squared_diffs":0},"commit_lat":{"mean":0.061673463,"squared_diffs":0},"idle_lat":{"mean":0,"squared_diffs":0},"bytes_read":{"mean":0,"squared_diffs":0},"rows_read":{"mean":0,"squared_diffs":0},"exec_stats":{"count":1,"network_bytes":{"mean":0,"squared_diffs":0},"max_mem_usage":{"mean":10240,"squared_diffs":0},"contention_time":{"mean":0,"squared_diffs":0},"network_messages":{"mean":0,"squared_diffs":0},"max_disk_usage":{"mean":0,"squared_diffs":0},"cpu_sql_nanos":{"mean":1719706,"squared_diffs":0},"mvcc_iterator_stats":{"step_count":{"mean":0,"squared_diffs":0},"step_count_internal":{"mean":0,"squared_diffs":0},"seek_count":{"mean":0,"squared_diffs":0},"seek_count_internal":{"mean":0,"squared_diffs":0},"block_bytes":{"mean":0,"squared_diffs":0},"block_bytes_in_cache":{"mean":0,"squared_diffs":0},"key_bytes":{"mean":0,"squared_diffs":0},"value_bytes":{"mean":0,"squared_diffs":0},"point_count":{"mean":0,"squared_diffs":0},"points_covered_by_range_tombstones":{"mean":0,"squared_diffs":0},"range_key_count":{"mean":0,"squared_diffs":0},"range_key_contained_points":{"mean":0,"squared_diffs":0},"range_key_skipped_points":{"mean":0,"squared_diffs":0}},"admission_wait_time":{"mean":0,"squared_diffs":0}},"rows_written":{"mean":0,"squared_diffs":0},"kv_cpu_time_nanos":{"mean":0,"squared_diffs":0}},"aggregated_ts":"2026-04-22T19:00:00Z","transaction_fingerprint_id":13371159268399818087,"aggregation_interval":0}}],"stmts_total_runtime_secs":15.303111,"txns_total_runtime_secs":5.2318454,"oldest_aggregated_ts_returned":"2026-04-22T19:00:00Z","stmts_source_table":"crdb_internal.statement_statistics","txns_source_table":"crdb_internal.transaction_statistics"}
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/bdcf2ab810bc38279e9dce2da8e9dabd/logTestStatusAPICombinedStatementsWithFullScans4161333895
--- FAIL: TestStatusAPICombinedStatementsWithFullScans (27.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusAPICombinedStatementsWithFullScans.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/server/application_api/sql_stats_test.go
+++ b/pkg/server/application_api/sql_stats_test.go
@@ -794,13 +794,20 @@ func TestStatusAPICombinedStatementsWithFullScans(t *testing.T) {
 		return string(bytes)
 	}
 
-	// Helper function to verify the combined statement statistics response.
-	verifyCombinedStmtStats := func() {
-		err := srvtestutils.GetStatusJSONProtoWithAdminAndTimeoutOption(firstServerProto, endpoint, &resp, false, additionalTimeout)
-		require.NoError(t, err)
+	// verifyCombinedStmtStats checks that the combined statement statistics
+	// response contains all expected statements with the correct metadata.
+	// Returns an error instead of failing the test directly so it can be
+	// retried via SucceedsSoon — the combined statements endpoint selects
+	// its source table (persisted vs. in-memory) based on data availability,
+	// and a background stats flush can cause the endpoint to read from the
+	// persisted table before newly executed statements have been persisted.
+	verifyCombinedStmtStats := func() error {
+		if err := srvtestutils.GetStatusJSONProtoWithAdminAndTimeoutOption(
+			firstServerProto, endpoint, &resp, false, additionalTimeout,
+		); err != nil {
+			return err
+		}
 
-		// actualResponseStatsMap maps the query response format to the actual
-		// statement statistics received from the server response.
 		actualResponseStatsMap := make(map[string]serverpb.StatementsResponse_CollectedStatementStatistics)
 		for _, respStatement := range resp.Statements {
 			actualResponseStatsMap[respStatement.Key.KeyData.Query] = respStatement
@@ -808,23 +815,36 @@ func TestStatusAPICombinedStatementsWithFullScans(t *testing.T) {
 
 		for respQuery, expectedData := range expectedStatementStatsMap {
 			respStatement, exists := actualResponseStatsMap[respQuery]
-			require.True(t, exists, "Expected statement '%s' not found in response: %v", respQuery, responseToJSON(resp))
+			if !exists {
+				return fmt.Errorf(
+					"expected statement '%s' not found in response: %v",
+					respQuery, responseToJSON(resp),
+				)
+			}
 
 			actualCount := respStatement.Stats.FirstAttemptCount
 			actualFullScan := respStatement.Key.KeyData.FullScan
 			actualDistSQL := respStatement.Key.KeyData.DistSQL
 
-			stmtJSONString := responseToJSON(respStatement)
-
-			require.Equal(t, expectedData.fullScan, actualFullScan, "failed for respStatement: %v", stmtJSONString)
-			require.Equal(t, expectedData.distSQL, actualDistSQL, "failed for respStatement: %v", stmtJSONString)
-			require.Equal(t, expectedData.count, int(actualCount), "failed for respStatement: %v", stmtJSONString)
+			if expectedData.fullScan != actualFullScan {
+				return fmt.Errorf("fullScan: expected %v, got %v for %s",
+					expectedData.fullScan, actualFullScan, respQuery)
+			}
+			if expectedData.distSQL != actualDistSQL {
+				return fmt.Errorf("distSQL: expected %v, got %v for %s",
+					expectedData.distSQL, actualDistSQL, respQuery)
+			}
+			if expectedData.count != int(actualCount) {
+				return fmt.Errorf("count: expected %d, got %d for %s",
+					expectedData.count, int(actualCount), respQuery)
+			}
 		}
+		return nil
 	}
 
 	// Execute and verify the queries that will be executed before the index is created.
 	executeStatements(statementsBeforeIndex)
-	verifyCombinedStmtStats()
+	testutils.SucceedsSoon(t, verifyCombinedStmtStats)
 
 	// Execute the queries that will create the index.
 	executeStatements(statementsCreateIndex)
@@ -849,7 +869,7 @@ func TestStatusAPICombinedStatementsWithFullScans(t *testing.T) {
 
 	// Execute and verify the queries that will be executed after the index is created.
 	executeStatements(statementsAfterIndex)
-	verifyCombinedStmtStats()
+	testutils.SucceedsSoon(t, verifyCombinedStmtStats)
 }
 
 func TestStatusAPICombinedStatements(t *testing.T) {

